Quantum-Control-Applications/Superconducting/Single Fixed Transmon/rb_without_singleshot_readout.py

```diff
@@ -0,0 +1,258 @@
+                    dual_demod.full("minus_sin", "out1", "cos", "out2", Q),
+                )
+
+                save(I, I_st)
+                save(Q, Q_st)
+
+                assign(sequence_list[depth], saved_gate)
+        save(n, n_st)
+
+    with stream_processing():
+        I_st.buffer(max_circuit_depth).buffer(num_of_sequences).average().save("I")
+        Q_st.buffer(max_circuit_depth).buffer(num_of_sequences).average().save("Q")
+
+qm = qmm.open_qm(config)
+
+job = qm.execute(rb)
+res_handles = job.result_handles
+
+I_handle = res_handles.get("I")
+Q_handle = res_handles.get("Q")
+iteration_handle = res_handles.get("iteration")
+I_handle.wait_for_values(1)
+Q_handle.wait_for_values(1)
+iteration_handle.wait_for_values(1)
+next_percent = 0.1  # First time print 10%
+
+x = np.linspace(1, max_circuit_depth, max_circuit_depth)
+
+
+def on_close(event):
+    event.canvas.stop_event_loop()
+    job.halt()
+
+
+f = plt.figure()
+f.canvas.mpl_connect("close_event", on_close)
+print("Progress =", end=" ")
+
+while res_handles.is_processing():
+    plt.cla()
+    I = np.average(I_handle.fetch_all(), axis=0)
+    Q = np.average(Q_handle.fetch_all(), axis=0)
+    iteration = iteration_handle.fetch_all()
+    if iteration / n_avg > next_percent:
+        percent = 10 * round(iteration / n_avg * 10)  # Round to nearest 10%
+        print(f"{percent}%", end=" ")
+        next_percent = percent / 100 + 0.1  # Print every 10%
+
+    plt.plot(x, I, ".", label="I")
+    plt.plot(x, Q, ".", label="Q")
+    plt.xlabel("Number of cliffords")
+
+    plt.legend()
+    plt.pause(0.1)
+
+plt.cla()
+I = np.average(I_handle.fetch_all(), axis=0)
+Q = np.average(Q_handle.fetch_all(), axis=0)
+iteration = iteration_handle.fetch_all()
+print(f"{round(iteration/n_avg * 100)}%")
+plt.plot(x, I, ".", label="I")
+plt.plot(x, Q, ".", label="Q")
+
+plt.legend()
+
+value = I  # Can change to Q
+
+
+def power_law(m, a, b, p):
+    return a * (p**m) + b
+
+
+pars, cov = curve_fit(
+    f=power_law,
+    xdata=x,
+    ydata=value,
+    p0=[0.5, 0.5, 0.9],
+    bounds=(-np.inf, np.inf),
+    maxfev=2000,
+)
+
+plt.plot(x, power_law(x, *pars), linestyle="--", linewidth=2)
+
+stdevs = np.sqrt(np.diag(cov))
+
+print("#########################")
+print("### Fitted Parameters ###")
+print("#########################")
+print(f"A = {pars[0]:.3} ({stdevs[0]:.1}), B = {pars[1]:.3} ({stdevs[1]:.1}), p = {pars[2]:.3} ({stdevs[2]:.1})")
+print("Covariance Matrix")
+print(cov)
+
+one_minus_p = 1 - pars[2]
+r_c = one_minus_p * (1 - 1 / 2**1)
+r_g = r_c / 1.875
+r_c_std = stdevs[2] * (1 - 1 / 2**1)
+r_g_std = r_c_std / 1.875
+
+print("#########################")
+print("### Useful Parameters ###")
+print("#########################")
+print(
+    f"1-p = {np.format_float_scientific(one_minus_p, precision=2)} ({stdevs[2]:.1}), "
+    f"r_c = {np.format_float_scientific(r_c, precision=2)} ({r_c_std:.1}), "
+    f"r_g = {np.format_float_scientific(r_g, precision=2)}  ({r_g_std:.1})"
+)
+
```

Comment:
Do you know what this is ?

Reply:
Yes, while they are known in the but maybe we should expand on these.
r_c is the Clifford infidelity.
1.875 is the number of average gates in the Clifford set (you can do RB with other sets), so r_g is the single gate infidelity.

Reply:
Yes it would be definitely useful for us and also some customers to add some comments explaining it (Fink's students did not know much about RB and I was not able to explain these numbers either...)